src/drtsans/iq.py

```diff
@@ -732,29 +738,40 @@ def _bin_iq1d(bin_edges, q_vec, dq_vec, i_vec, error_vec):
         Returns
         -------
         ~tuple
-            binned intensity vector, binned intensity error vector, binned q resolution vector
+            binned intensity vector,
+            binned intensity error vector,
+            binned q resolution vector,
+            binned q center vector
 
         """
+
         # Count number of Q in 'q_array' in each Q-bin when they are binned (histogram) to 'bin_edges'
-        num_pt_vec, _ = np.histogram(q_vec, bins=bin_edges)
+        num_pt_vec, _ = np.histogram(q_vec, bins=q_bins.edges)
 
         # Counts per bin: I_{k, raw} = \sum I(i, j) for each bin
-        i_raw_vec, _ = np.histogram(q_vec, bins=bin_edges, weights=i_vec)
+        i_raw_vec, _ = np.histogram(q_vec, bins=q_bins.edges, weights=i_vec)
 
         # Square of summed uncertainties for each bin
-        sigma_sqr_vec, _ = np.histogram(q_vec, bins=bin_edges, weights=error_vec**2)
+        sigma_sqr_vec, _ = np.histogram(q_vec, bins=q_bins.edges, weights=error_vec**2)
 
-        # Final I(Q):     I_k       = \frac{I_{k, raw}}{N_k}
+        all_lists = iter([num_pt_vec, i_raw_vec, sigma_sqr_vec])
+        assert all(len(lst) == len(num_pt_vec) for lst in all_lists)
+
+        # Final I(Q):     I_k = \frac{I_{k, raw}}{N_k}
         i_final_vec = i_raw_vec / num_pt_vec
-        # Final sigma(Q): sigmaI_k  = \frac{sigmaI_{k, raw}}{N_k}
+
+        # Final sigma(Q): sigmaI_k = \frac{sigmaI_{k, raw}}{N_k}
         sigma_final_vec = np.sqrt(sigma_sqr_vec) / num_pt_vec
+        # Replace 0 error with 1, as uncertainty is not zero in the region of interest
+        sigma_final_vec[sigma_final_vec == 0] = 1.0
 
         # Calculate Q resolution of binned
         if dq_vec is None:
             bin_dq_vec = None
         else:
-            binned_vec, _ = np.histogram(q_vec, bins=bin_edges, weights=dq_vec)
+            binned_vec, _ = np.histogram(q_vec, bins=q_bins.edges, weights=dq_vec)
             bin_dq_vec = binned_vec / num_pt_vec
+            assert len(bin_dq_vec) == len(i_final_vec)
 
         return i_final_vec, sigma_final_vec, bin_dq_vec
```

Comment:
For the  triad `i_final_vec, sigma_final_vec, bin_dq_vec`, we need to flag as `np.nan` those indexes where `num_pt_vec` is zero because for these arrays should contain the value `0 / 0` for these indexes.